### PR TITLE
Nested route/fragment support

### PR DESCRIPTION
**Pull request by tptee**
- [x] Support nested route definitions by flattening them at configuration time
- [x] Attach parent route/results to flat route definitions
- [x] Add `RelativeFragment`
- [x] Add tests and demo for nested Fragments
/cc @baer @dpwrussell

### CODE REVIEW

**Comment by tptee** [comment]
This is ready for review while I update the docs /cc @baer @divmain

File: README.md [hunk 1]
```diff
@@ -197,23 +210,65 @@ The simplest fragment is one that displays when a route is active:
...
+```
+
+`<RelativeFragment>` lets you nest fragments to match your UI hierarchy to your route hierarchy, much like the `<Route>` component does in `react-router`. Given a URL of `/home/bio/ken-m`, and the following elements:
+
+```js
+<RelativeFragment forRoute='/home'>
+  <h1>Home</h1>
+  <RelativeFragment forRoute='/bio'>
+    <h2>Bios</h2>
+    <RelativeFragment forRoute='/dat-boi'>
```
Comment (kriuz) [inline README.md]: Should this be `<RelativeFragment forRoute='/ken-m'>` ?

Comment (tptee) [inline README.md]: Sure should! Thanks for catching that 👍

File: README.md [hunk 2]
```diff
@@ -197,23 +210,65 @@ The simplest fragment is one that displays when a route is active:
 </Fragment>
 ```
 
-You can also specify a fragment that displays on multiple routes:
+You can also match a fragment against anything in the current `location` object:
 
 ```es6
-<Fragment forRoutes={['/home/messages', '/home']}>
-  <p>This displays in a couple of places!</p>
+<Fragment withConditions={location => location.query.superuser}>
```
Comment (divmain) [inline README.md]: That's very cool!

**Comment by divmain** [comment]
Nested routes look 💯

**Pull request merged**